File: mydagger2/src/main/java/com/zhongjh/mydagger2/module/MainActivity.java
package com.zhongjh.mydagger2.module;

import android.os.Bundle;
import android.widget.TextView;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

import com.zhongjh.mydagger2.R;

import javax.inject.Inject;

/**
 * @author zhongjh
 * @date 2022/4/24
 */
public class MainActivity extends AppCompatActivity {

    @Inject
    MainParameter mainParameter;

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        DaggerMainComponent.builder().mainModule(new MainModule("参数")).build().inject(this);

        ((TextView) findViewById(R.id.tvName)).append("_" + mainParameter.getName());
    }


}

File: myhilt/src/main/java/com/zhongjh/hiltdemo/havearguments/MainActivity.kt
package com.zhongjh.hiltdemo.havearguments

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.zhongjh.hiltdemo.databinding.ActivityMainBinding
import dagger.hilt.android.AndroidEntryPoint
import javax.inject.Inject

/**
 * 这是演示一个 有参数 的构造函数
 */
@AndroidEntryPoint
class MainActivity : AppCompatActivity() {

    @Inject
    lateinit var mainFactory: MainFactory
    lateinit var mainEntity: MainEntity

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        val binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)
        mainEntity = mainFactory.create("名称", "地区")
        binding.tvName.append(mainEntity.name)
        binding.tvName.append(mainEntity.area)
    }
}